sensor/common/deploymentenhancer/component.go

```diff
@@ -0,0 +1,144 @@
+	go func() {
+		for {
+			deploymentMsg, more := <-d.deploymentsQueue
+			if !more {
+				return
+			}
+			requestID := deploymentMsg.GetMsg().GetId()
+			if requestID == "" {
+				log.Warnf("received deploymentEnhancement msg with empty request ID. Discarding request.")
+				continue
+			}
+			deployments := d.extractAndEnrichDeployments(deploymentMsg)
+			d.sendDeploymentsToCentral(requestID, deployments)
+		}
+	}()
+	return nil
+}
+
+func (d *DeploymentEnhancer) extractAndEnrichDeployments(deploymentMsg *central.DeploymentEnhancementRequest) []*storage.Deployment {
+	var ret []*storage.Deployment
+
+	deployments := deploymentMsg.GetMsg().GetDeployments()
+	if deployments == nil {
+		log.Warnf("received deploymentEnhancement msg with no deployments")
+		return ret
+	}
+
+	log.Debugf("Received deploymentEnhancement msg with %v deployment(s)", len(deploymentMsg.GetMsg().GetDeployments()))
+	for _, deployment := range deployments {
+		enriched, err := d.enrichDeployment(deployment)
+		if err != nil {
+			log.Warnf("Failed to enrich deployment: %v", deployment)
+			continue
+		}
+		ret = append(ret, enriched)
+	}
+
+	return ret
+}
+
+func (d *DeploymentEnhancer) sendDeploymentsToCentral(id string, deployments []*storage.Deployment) {
+	d.responsesC <- message.New(&central.MsgFromSensor{
+		Msg: &central.MsgFromSensor_DeploymentEnhancementResponse{
+			DeploymentEnhancementResponse: &central.DeploymentEnhancementResponse{
+				Msg: &central.DeploymentEnhancementMessage{
+					Id:          id,
+					Deployments: deployments,
+				},
+			},
+		},
+	})
+}
+
+func (d *DeploymentEnhancer) enrichDeployment(deployment *storage.Deployment) (*storage.Deployment, error) {
+	localImages := d.findLocalImages(deployment)
+
+	p := d.storeProvider.RBAC().GetPermissionLevelForDeployment(deployment)
+	e := d.storeProvider.Services().GetExposureInfos(deployment.GetNamespace(), deployment.GetPodLabels())
+
+	deployment = d.storeProvider.Deployments().EnhanceDeploymentReadOnly(deployment, store.Dependencies{
+		PermissionLevel: p,
+		Exposures:       e,
+		LocalImages:     localImages,
+	})
+
+	return deployment, nil
+}
+
+func (d *DeploymentEnhancer) findLocalImages(deployment *storage.Deployment) set.Set[string] {
+	localImages := set.NewStringSet()
+	for _, c := range deployment.GetContainers() {
+		imgName := c.GetImage().GetName()
+		if d.storeProvider.Registries().IsLocal(imgName) {
+			localImages.Add(imgName.GetFullName())
+		}
+	}
+	return localImages
+}
+
+// Capabilities return the capabilities of this component
+func (d *DeploymentEnhancer) Capabilities() []centralsensor.SensorCapability {
+	// TODO(ROX-21197): Add Capability
+	return nil
+}
+
+// ResponsesC returns the response channel of this component
+func (d *DeploymentEnhancer) ResponsesC() <-chan *message.ExpiringMessage {
+	return d.responsesC
+}
+
+// Stop stops the component
+func (d *DeploymentEnhancer) Stop(_ error) {
+	defer close(d.deploymentsQueue)
+}
+
+// Notify .
```

Comment:
Don't forget to add a proper comment here.

Reply:
Done!